Content.Server/Weapons/Melee/Telebaton/TelebatonComponent.cs

```diff
@@ -0,0 +1,23 @@
+using Content.Server.Stunnable.Systems;
+using Content.Shared.Timing;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Weapons.Melee.Telebaton
+{
+  [RegisterComponent, Access(typeof(TelebatonSystem))]
+  public sealed class TelebatonComponent : Component 
+  {
+    [DataField("activated")]
+    public bool Activated { get; set; } = false;
+
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("onThrowStunChance")]
+    public float onThrowStunChance { get; set; } = 0.20f;
+
+    [DataField("stunSound")]
+    public SoundSpecifier StunSound { get; set; } = new SoundPathSpecifier("/Audio/Weapons/egloves.ogg");
+
+    [DataField("extendSound")]
+    public SoundSpecifier ExtendSound { get; set; } = new SoundPathSpecifier("/Audio/Weapons/batonextend.ogg");
+  }
+}
```

Comment:
Components shouldn't be singleton.
1. Stun baton, telebaton, and esword should be using a generic toggleable component.
2. Stun baton and telebaton should be using the same code.

Reply:
will do.

Reply:
i tried combining it all in one component, but it seems that better solution is just let it be like it been i believe.